editor/src/tool-bar/tool-bar.ts

```diff
@@ -113,7 +121,7 @@ export class ToolBar extends AbstractUIExtension implements IActionHandler, Edit
 
     const toolBarButtons = this.toolBarButtonProvider
       .map(provider => provider.button(() => [...this.selectionService.getSelectedElementIDs()]))
-      .filter(button => !button?.id)
+      .filter(button => !button?.id && !button?.location)
```

Comment:
Simply check the location here is right, (and make the location strict, not nullable). 
And add locations also to the other buttons. 
And change also the other filters.